week 6.cpp

```diff
@@ -0,0 +1,204 @@
+#include <iostream>
+#include <stdlib.h>
+using namespace std;
+int id=0;
+class Item
+{
+	int item_id;
+	//char item_name;
+	float price;
+	
+	
+	public:
+
+	int item_count;
+	Item()
+	{
+		item_id=id++;
+		price=10;
+		item_count=2;
+		
+	}
+	float get_price()
+	{
+	   return price;
+	}
+
+	void decrease_item_count()
+	{
+
+		item_count--;
+
+	}
+
+	void increase_item_count()
+	{
+
+		item_count++;
+
+	}
+
+
+
+};
+
+class coin
+{
+  int count_1; // no of 1 rupee coins
+  int count_5;  // no of 5 rupee coins
+  int count_10; // no of 10 rupee coins
+  
+  public:
+  coin()
+  {
+  	count_1=20;
+  	count_5=10;
+  	count_10=3;
+  }
+	
+};
+
+class Vending_Machine
+{
+	float current_balance;  // current_balance = count_1 + count_5 + count_10
+	int available_items[3];
```

Comment:
3 is a magic number used in several places throughout this class.  While I don't think this should be a hard coded value at all, if you do want to have a fixed number this should be a constant so that you can update all of it's uses in one place

Reply:
The 3 should probably be a parameter passed in to the vending machine